tensorflow/core/common_runtime/bfc_allocator.cc

```diff
@@ -178,7 +178,8 @@ bool BFCAllocator::Extend(size_t alignment, size_t rounded_bytes) {
 }
 
 BFCAllocator::ChunkHandle BFCAllocator::AllocateChunk() {
-  if (free_chunks_list_ != kInvalidChunkHandle) {
+  const int free_chunks_list_int = free_chunks_list_;
+  if (free_chunks_list_int != kInvalidChunkHandle) {
```

Comment:
Continuing discussion from https://github.com/tensorflow/tensorflow/pull/40897#discussion_r449783084 you can safely have

```
const uint64_t = -1;
```

It will just be a constant of 64 bits of 1. There's no functionality change, but changing the type of the constant is better for readability of the code.

Reply:
Okay, just for clarity, change `kInvalidChunkHandle` to `const uint64_t`?

Reply:
I was just giving an example. Here it should be `ChunkHandle`

Reply:
Here I am recasting `free_chunks_list_` to `int` to avoid warning when it is compared to `kInvalidChunkHandle`. Can you clarify the change you would like me to make? 😅

Reply:
You have multiple `static_cast<int>(h)` in this file to match the type of `kInvalidChunkHandle`. Instead of doing the cast, we can change the type of the constant.

https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/core/common_runtime/bfc_allocator.h;l=135-136;drc=59283c98c9abdedd34657d326cec9ce10460c987

Reply:
`const uint64_t kInvalidChunkHandle = -1`?

Reply:
`const ChunkHandle kInvalidChunkHandle = -1`

Reply:
Ok. Thank you.

Reply:
I'm removing the casts now. Could you say a bit more about why making kInvalidChunkHandle unsigned is safe, i.e. wont cause unintended comparisons?

Reply:
Most comparisons are between a `ChunkHandle h` and `kInvalidChunkHandle`. Thus, both should have the same type.

Spurious comparisons between `int x` and `kInvalidChunkHandle` will need a cast, but these should be extremely rare now.